src/python_testing/TC_OPCREDS_3_2.py

```diff
@@ -0,0 +1,79 @@
+#
+#    Copyright (c) 2024 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import logging
+import random
+
+import chip.clusters as Clusters
+from chip.utils import CommissioningBuildingBlocks
+from chip import ChipDeviceCtrl
+from matter_testing_support import MatterBaseTest, async_test_body, default_matter_test_main
+from mobly import asserts
+
+
+class TC_OPCREDS_3_2(MatterBaseTest):
+    # All global functions
+    @async_test_body
+    async def test_TC_OPCREDS_3_2(self):
+
+        
+
+        # It is not necessary implement factory reset to DUT
+        self.print_step(1, "Step 1: Factory Reset DUT")
+
+        dev_ctrl = self.default_controller
+
+        new_certificate_authority = self.certificate_authority_manager.NewCertificateAuthority()
+        th1_new_fabric_admin = new_certificate_authority.NewFabricAdmin(vendorId=0xFFF1, fabricId=2222)
+        th1_nodeid = self.default_controller.nodeId+1
+        th1_dut_nodeid = self.dut_node_id+1
+
+        th1_new_admin_ctrl = th1_new_fabric_admin.NewController(nodeId=th1_nodeid)
+        resp = await CommissioningBuildingBlocks.AddNOCForNewFabricFromExisting(
+            commissionerDevCtrl=dev_ctrl, newFabricDevCtrl=th1_new_admin_ctrl,
+            existingNodeId=self.dut_node_id, newNodeId=th1_dut_nodeid
+        )
+
+        self.print_step(2, "Commission DUT to TH1's Fabric")
+
+        # CommissioningBuildingBlocks just return a boolean. I need to get the NOC Response
```

Comment:
This function is entirely implemented in python, you can just return the stuff you need from that function - it's all available. just return a tuple.

Reply:
Yes, I modified both. I added two new values to return in commissioningBuildingBlocks (rcac and noc response).